src/main/java/no/digipost/signature/client/core/internal/ClientHelper.java

```diff
@@ -99,37 +95,41 @@ XMLPortalSignatureJobResponse call() {
+            public XMLDirectSignatureJobStatusResponse call() {
                 Response response = httpClient.target(statusUrl)
                         .request()
                         .accept(APPLICATION_XML_TYPE)
                         .get();
-                Status status = Status.fromStatusCode(response.getStatus());
-                if (status == OK) {
-                    return response.readEntity(XMLDirectSignatureJobStatusResponse.class);
-                } else if (status == FORBIDDEN) {
-                    XMLError error = extractError(response);
-                    if (ErrorCodes.INVALID_STATUS_QUERY_TOKEN.sameAs(error.getErrorCode())) {
-                        throw new InvalidStatusQueryTokenException(statusUrl, error.getErrorMessage());
-                    }
-                } else if (status == NOT_FOUND) {
-                    XMLError error = extractError(response);
-                    if (SIGNING_CEREMONY_NOT_COMPLETED.sameAs(error.getErrorCode())) {
-                        throw new CantQueryStatusException(status, error.getErrorMessage());
+                try {
+                    Status status = Status.fromStatusCode(response.getStatus());
+                    if (status == OK) {
+                        return response.readEntity(XMLDirectSignatureJobStatusResponse.class);
+                    } else if (status == FORBIDDEN) {
+                        XMLError error = extractError(response);
+                        if (ErrorCodes.INVALID_STATUS_QUERY_TOKEN.sameAs(error.getErrorCode())) {
+                            throw new InvalidStatusQueryTokenException(statusUrl, error.getErrorMessage());
+                        }
+                    } else if (status == NOT_FOUND) {
+                        XMLError error = extractError(response);
+                        if (SIGNING_CEREMONY_NOT_COMPLETED.sameAs(error.getErrorCode())) {
+                            throw new CantQueryStatusException(status, error.getErrorMessage());
+                        }
                     }
+                    throw exceptionForGeneralError(response);
+                } finally {
+                    response.close();
                 }
-                throw exceptionForGeneralError(response);
             }
         });
     }
 
-    public InputStream getSignedDocumentStream(final String url) {
-        return call(new Producer<InputStream>() {
+    public InputStream getSignedDocumentStream(final String uri) {
+        return call(new Callable<InputStream>() {
             @Override
-            InputStream call() {
-                return parseResponse(httpClient.target(url).request().accept(APPLICATION_XML_TYPE, APPLICATION_OCTET_STREAM_TYPE).get(), InputStream.class);
+            public InputStream call() {
+                return parseResponse(httpClient.target(uri).request().accept(APPLICATION_XML_TYPE, APPLICATION_OCTET_STREAM_TYPE).get(), InputStream.class);
```

Comment:
This returned InputStream will also close the Response when the stream is closed. This is (as of now) the only place were we leave the responsibility of closing resources on the user of the client library.